### DISCUSSION TITLE
Exceptions in Replace Child

### DISCUSSION THREAD

**Discussion by What-is-the-code** [community/community#21421] [Other Feature Feedback, Questions, & Ideas]
<p>Hello<br>
Can anyone explain this exception in replace-child DOM “If the replacement of oldchild by newchild would lead to a cycle, that is if newchild is an ancestor of the node”<br>
with example?</p>
<p>Thanks</p>

**Comment by billythekid** [comment]
<p>Hi <a class="mention" href="/what-is-the-code">@What-is-the-code</a> that’s a good question!</p>
<p>You’re referring to the exception thrown by the WebAPI <code>replaceChild</code> method:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://developer.mozilla.org/en-US/docs/Web/API/Node/replaceChild">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/330170/181064504-984b2225-88cb-4344-afb1-de0dd656a0dc.png" class="site-icon" width="48" height="48">

      <a href="https://developer.mozilla.org/en-US/docs/Web/API/Node/replaceChild" target="_blank" rel="noopener">developer.mozilla.org</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/388;"><img src="https://user-images.githubusercontent.com/330170/181064505-c0bbecc1-2443-4842-aef1-6291c254ba69.png" class="thumbnail" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/7/c/7c212d669f9e3367480797f5d5bf0d81e72e5573_2_690x388.png, https://aws1.discourse-cdn.com/github/optimized/3X/7/c/7c212d669f9e3367480797f5d5bf0d81e72e5573_2_1035x582.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/7/c/7c212d669f9e3367480797f5d5bf0d81e72e5573_2_1380x776.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/c/7c212d669f9e3367480797f5d5bf0d81e72e5573_2_10x10.png"></div>

<h3><a href="https://developer.mozilla.org/en-US/docs/Web/API/Node/replaceChild" target="_blank" rel="noopener">Node.replaceChild() - Web APIs | MDN</a></h3>

  <p>The replaceChild() method of the Node element
  replaces a child node within the given (parent) node.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>It might be helpful to think of it like this.</p>
<p>If I have a <code>div</code> which contains an <code>img</code> and I want to replace the image with something else, I can’t replace it with the <code>div</code> because the <code>img</code> itself is still there. These aren’t copies of the elements they are the actual elements themselves so that <code>img</code> hasn’t been replaced at all.</p>
<p>Here’s a <a href="https://codepen.io/billythekid/pen/LYzOvpK">working CodePen</a> showing exactly this concept.</p>
<p>Hope that helps.</p>
<p>Cheers,<br>
Billy :cowboy_hat_face:</p>

**Comment by What-is-the-code** [comment]
<p>Thanks <a class="mention" href="/billythekid">@billythekid</a> Got it<br>
But would really appreciate if YOU fit in the above line in it as well<br>
Mean what in the example is oldchild and what is newchild for more deep understanding.</p>
